packages/devtools_app/lib/src/app.dart

```diff
@@ -171,35 +195,42 @@ class DevToolsAppState extends State<DevToolsApp> {
       url: vmServiceUri,
       allowConnectionScreenOnDisconnect: !embed,
       builder: (_) {
-        final tabs = _visibleScreens()
-            .where((p) => embed && page != null ? p.screenId == page : true)
-            .where((p) => !hide.contains(p.screenId))
-            .toList();
-        if (tabs.isEmpty) {
-          return DevToolsScaffold.withChild(
-            child: CenteredMessage(
-                'The "$page" screen is not available for this application.'),
-            ideTheme: ideTheme,
-            analyticsProvider: widget.analyticsProvider,
-          );
-        }
-        return _providedControllers(
-          child: DevToolsScaffold(
-            embed: embed,
-            ideTheme: ideTheme,
-            page: page,
-            tabs: tabs,
-            analyticsProvider: widget.analyticsProvider,
-            actions: [
-              // TODO(https://github.com/flutter/devtools/issues/1941)
-              if (serviceManager.connectedApp.isFlutterAppNow) ...[
-                HotReloadButton(),
-                HotRestartButton(),
-              ],
-              OpenSettingsAction(),
-              OpenAboutAction(),
-            ],
-          ),
+        // Force regeneration of visible screens when VM developer mode is
+        // enabled.
+        return ValueListenableBuilder<bool>(
```

Comment:
can you remove this value listenable builder now? if the routes need to be regenerated, try calling clearCachedRoutes in setState call from the vmDeveloperModeEnabled listener above

Reply:
No, that doesn't seem to work.